server/game/cards/01_SOR/leaders/BobaFettCollectingTheBounty.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+
+export default class BobaFettCollectingTheBounty extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '4626028465',
+            internalName: 'boba-fett#collecting-the-bounty',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'Ready a resource',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            when: {
+                onCardReturnedToHand: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller && context.source.controller.resources.some((resource) => resource.exhausted),
```

Comment:
I just realized we don't have it yet, but we definitely need an "onLeavesPlay" event that happens in parallel with any card being defeated / returned to hand from play.

If you're willing to tackle it that'd be awesome, it should be pretty straightforward but the place to add it is not super obvious. If you're up for it, instructions are below, otherwise just ping me and I can do a patch in the next day or two.

We have a concept of "contingent events" which are events that get emitted if-and-only-if a prior event actually resolves. Current examples are (1) overwhelm damage, which only happens if the original damage instance isn't prevented by a shield, and (2) defeating upgrades on a unit leaving play.

The code location for (2) above is where the "onLeavesPlay" event should also be emitted (note you will need add a new entry for it in Constants.ts). 

Go to `CardTargetSystem.addLeavesPlayPropertiesToEvent()` and update it to emit an anonymous event of type 'onLeavesPlay.` For an example of how to create such an event, see here: https://github.com/SWU-Karabast/forceteki/blob/22a2a2a5cb1565efd74d000fb29bcd8dc747b37d/server/game/core/gameSteps/AbilityResolver.js#L78